### PR TITLE
Fix Issue 811 for QoL

### PR DESCRIPTION
**Pull request by JonathanYiv**
AJAX for lesson completion on course page
#811

### CODE REVIEW

File: app/views/courses/course/_section-lessons.html.erb [hunk 1]
```diff
@@ -6,9 +6,9 @@
         <%= numbered_lesson_title(lesson, lesson_index) %>
       <% end %>
 
-      <% if user_signed_in? %>
-          <i class="fa fa-check-circle-o section-lessons__item__icon <%= lesson_completed?(@user, lesson) %>"></i>
-      <% end %>
+      <div id="section-lessons__<%= lesson.id %>">
```
Comment (KevinMulhern) [inline app/views/courses/course/_section-lessons.html.erb]: can we wrap this div in a check to make sure the user exists before rendering it please?
 `<% if user_signed_in? %>`

I know you have that in the partial already, the two levels of if statements in that file make it a bit difficult to read though.

File: app/views/lesson_completions/create.js.erb [hunk 2]
```diff
@@ -1,2 +1,7 @@
-$('.lesson-button-group').html("<%= j render 'lessons/lesson_buttons', lesson: @lesson, course: @lesson.course, user: @user  %>");
 $('.banner__badge').html("<%= j render 'shared/course_badge', course: @lesson.course, user: @user, modifier: '' %>");
+
+<% if @came_from_lesson %>
```
Comment (KevinMulhern) [inline app/views/lesson_completions/create.js.erb]: I think we should depend on the DOM to determine what to render here, something like 
`if $('.lesson-button-group').length > 0`

There might be a better way of doing this though, @arun1595 will surely know 😄

**Comment by KevinMulhern** [comment]
This is looking really good @JonathanYiv, thanks for putting it together. If you have any questions or need any clarification  on any of my comments don't hesitate to let me know 😄

**Comment by JonathanYiv** [comment]
Thanks for the feedback on cleaner code!

**Comment by KevinMulhern** [comment]
Thanks for making those changes @JonathanYiv 
LGTM 👍

**Pull request merged**